### DISCUSSION TITLE
Getting multiline_textbbox without an image

### DISCUSSION THREAD

**Discussion by awhitefox** [python-pillow/Pillow#6655] [Q&A] [answered]
Hello, I have a question about calculating text bounding boxes.

**Is there a method that can calculate a bounding box of a given text without an actual image object?** The problem I having is that I need to have image size calculated beforehand. And to do this I need to get size of multiline text. So to create an image I need its size and to calculate the size I need to use `multiline_textbbox(...)` which is a method of `ImageDraw` which is instantiated using an image object.

**Solutions I've already tried:**

- Using `ImageFont.getbbox(...)` which produces a wrong result (height is ignored)
```python
import os
from PIL import ImageFont

font = ImageFont.truetype(os.path.join('fonts', 'OpenSans-Regular.ttf'), 14)
print(font.getbbox('line 1'))  # (0, 4, 37, 15)
print(font.getbbox('line 1\nline 2\nline 3'))  # (0, 4, 127, 15)
```

- Using `ImageFont.getsize_multiline(...)` which works but produces a deprecation warning
```python
import os
from PIL import ImageFont

font = ImageFont.truetype(os.path.join('fonts', 'OpenSans-Regular.ttf'), 14)
print(font.getsize_multiline('line 1\nline 2\nline 3'))  # (37, 53)

# DeprecationWarning: getsize_multiline is deprecated and will be removed in Pillow 10 (2023-07-01). Use ImageDraw.multiline_textbbox instead.
```

- Using a dummy image object **which works but feels hacky**
```python
import os

from PIL import Image
from PIL import ImageFont
from PIL.ImageDraw import ImageDraw


font = ImageFont.truetype(os.path.join('fonts', 'OpenSans-Regular.ttf'), 14)
draw = ImageDraw(Image.new('RGB', (0, 0)))
print(draw.multiline_textbbox((0, 0), 'line 1', font))  # (0, 4, 37, 15)
print(draw.multiline_textbbox((0, 0), 'line 1\nline 2\nline 3', font))  # (0, 4, 37, 53)
```

**Am I missing something or is there really no way to achieve this? And if there isn't why an image is required to calculate a size of text?**

**Comment by nulano** [comment, marked as answer]
> Using a dummy image object which works but feels hacky

This is currently the best option.

All multiline text operations, except for `font.getsize_multiline` (IIRC an exact duplicate of `draw.multiline_textsize`), are handled in `ImageDraw` which requires an Image. Moving the `draw.multiline_text` operation to `ImageFont` would require significant changes (either rewriting the function in C or taking a significant performance penalty), and `draw.multiline_textbbox` is quite long (~100 lines), too long to feel good about duplicating the code in `ImageFont`.
